### PR TITLE
Btcex: fetchLeverage, fetchMarketLeverageTiers, fetchLeverageTiers

### PR DESCRIPTION
**Pull request by Dan-krm**
Added fetchLeverage, fetchMarketLeverageTiers and fetchLeverageTiers to Btcex:
### fetchLeverage:
```
btcex.fetchLeverage (BTC/USDT:USDT)
2023-01-23T03:53:59.253Z iteration 0 passed in 5274 ms

20
```

### fetchMarketLeverageTiers:
```
btcex.fetchMarketLeverageTiers (BTC/USDT:USDT)
2023-01-23T03:57:08.206Z iteration 0 passed in 2647 ms

tier | currency | minNotional | maxNotional | maintenanceMarginRate | maxLeverage
---------------------------------------------------------------------------------
   1 |     USDT |           0 |       50000 |                 0.004 |         125
   2 |     USDT |       50000 |      250000 |                 0.005 |         100
   3 |     USDT |      250000 |     1000000 |                  0.01 |          50
   4 |     USDT |     1000000 |     7500000 |                 0.025 |          20
   5 |     USDT |     7500000 |    40000000 |                  0.05 |          10
   6 |     USDT |    40000000 |   100000000 |                   0.1 |           5
   7 |     USDT |   100000000 |   200000000 |                 0.125 |           4
   8 |     USDT |   200000000 |   400000000 |                  0.15 |           3
   9 |     USDT |   400000000 |   600000000 |                  0.25 |           2
  10 |     USDT |   600000000 |  1000000000 |                   0.5 |           1
10 objects
```

### fetchLeverageTiers:
```
btcex.fetchLeverageTiers (BTC/USDT:USDT)
2023-01-23T05:14:41.466Z iteration 0 passed in 3017 ms

{
  'BTC/USDT:USDT': [
    {
      tier: 1,
      currency: 'USDT',
      minNotional: 0,
      maxNotional: 50000,
      maintenanceMarginRate: 0.004,
      maxLeverage: 125,
      info: {
        bracket: '1',
        initialLeverage: '125',
        maintenanceMarginRate: '0.004',
        notionalCap: '50000',
        notionalFloor: '0',
        cum: '0'
      }
    },
    {
      tier: 2,
      currency: 'USDT',
      minNotional: 50000,
      maxNotional: 250000,
      maintenanceMarginRate: 0.005,
      maxLeverage: 100,
      info: {
        bracket: '2',
        initialLeverage: '100',
        maintenanceMarginRate: '0.005',
        notionalCap: '250000',
        notionalFloor: '50000',
        cum: '50'
      }
    },
    {
      tier: 3,
      currency: 'USDT',
      minNotional: 250000,
      maxNotional: 1000000,
      maintenanceMarginRate: 0.01,
      maxLeverage: 50,
      info: {
        bracket: '3',
        initialLeverage: '50',
        maintenanceMarginRate: '0.01',
        notionalCap: '1000000',
        notionalFloor: '250000',
        cum: '1300'
      }
    },
...
  ]
}
```

### CODE REVIEW

File: js/btcex.js [hunk 1]
```diff
@@ -1884,6 +1890,191 @@ module.exports = class btcex extends Exchange {
...
+        //
+        const tiers = {};
+        const result = {};
+        const marketIds = Object.keys (response);
+        for (let i = 0; i < marketIds.length; i++) {
+            const marketId = marketIds[i];
+            const entry = response[marketId];
+            const market = this.safeMarket (marketId);
+            const symbol = this.safeSymbol (marketId, market);
+            let symbolsLength = 0;
```
Comment (sc0Vu) [inline js/btcex.js:2063]: How about `const symbolsLength = symbols.length`?

File: js/btcex.js [hunk 2]
```diff
@@ -1884,6 +1890,191 @@ module.exports = class btcex extends Exchange {
...
+            tiers[symbol] = this.parseMarketLeverageTiers (entry, market);
+            if (symbols !== undefined) {
+                symbolsLength = symbols.length;
+                if (this.inArray (symbol, symbols)) {
+                    result[symbol] = this.parseMarketLeverageTiers (entry, market);
+                }
+            }
+            if (symbol !== undefined && (symbolsLength === 0 || this.inArray (symbols, symbol))) {
```
Comment (sc0Vu) [inline js/btcex.js:2071]: Does this work the same as the code above?

```JS
            if (symbols !== undefined) {
                if (this.inArray (symbol, symbols)) {
                  // ...
                }
            }
```

File: js/btcex.js (hunk 1 above)
Comment (Dan-krm) [reply js/btcex.js:2063]:
```
btcex.fetchLeverageTiers ()
[TypeError] Cannot read properties of undefined (reading 'length')
```

File: js/btcex.js (hunk 2 above)
Comment (Dan-krm) [reply js/btcex.js:2071]: Not for fetching all leverage tiers:
```
btcex.fetchLeverageTiers ()
{}
```

**Comment by carlosmiei** [comment]
LGTM ✅ 
```
p btcex fetchLeverage "BTC/USDT:USDT"
Python v3.10.8
CCXT v2.6.58
btcex.fetchLeverage(BTC/USDT:USDT)
20.0
```

```
 p 'btcex.fetchLeverageTiers(["BTC/USDT:USDT"])'
Python v3.10.8
CCXT v2.6.58
btcex.fetchLeverageTiers(['BTC/USDT:USDT'])
{'BTC/USDT:USDT': [{'currency': 'USDT',
                    'info': {'bracket': '1',
                             'cum': '0',
                             'initialLeverage': '125',
                             'maintenanceMarginRate': '0.004',
                             'notionalCap': '50000',
                             'notionalFloor': '0'},
                    'maintenanceMarginRate': 0.004,
                    'maxLeverage': 125.0,
                    'maxNotional': 50000.0,
                    'minNotional': 0.0,
                    'tier': 1},
                   {'currency': 'USDT',
                    'info': {'bracket': '2',
                             'cum': '50',
                             'initialLeverage': '100',
                             'maintenanceMarginRate': '0.005',
                             'notionalCap': '250000',
                             'notionalFloor': '50000'},
                    'maintenanceMarginRate': 0.005,
                    'maxLeverage': 100.0,
                    'maxNotional': 250000.0,
                    'minNotional': 50000.0,
                    'tier': 2},
                   {'currency': 'USDT',
                    'info': {'bracket': '3',
                             'cum': '1300',
                             'initialLeverage': '50',
                             'maintenanceMarginRate': '0.01',
                             'notionalCap': '1000000',
                             'notionalFloor': '250000'},
                    'maintenanceMarginRate': 0.01,
                    'maxLeverage': 50.0,
                    'maxNotional': 1000000.0,
                    'minNotional': 250000.0,
                    'tier': 3},
                   {'currency': 'USDT',
                    'info': {'bracket': '4',
                             'cum': '16300',
                             'initialLeverage': '20',
                             'maintenanceMarginRate': '0.025',
                             'notionalCap': '7500000',
                             'notionalFloor': '1000000'},
                    'maintenanceMarginRate': 0.025,
                    'maxLeverage': 20.0,
                    'maxNotional': 7500000.0,
                    'minNotional': 1000000.0,
                    'tier': 4},
                   {'currency': 'USDT',
                    'info': {'bracket': '5',
                             'cum': '203800',
                             'initialLeverage': '10',
                             'maintenanceMarginRate': '0.05',
                             'notionalCap': '40000000',
                             'notionalFloor': '7500000'},
                    'maintenanceMarginRate': 0.05,
                    'maxLeverage': 10.0,
                    'maxNotional': 40000000.0,
                    'minNotional': 7500000.0,
                    'tier': 5},
                   {'currency': 'USDT',
                    'info': {'bracket': '6',
                             'cum': '2203800',
                             'initialLeverage': '5',
                             'maintenanceMarginRate': '0.1',
                             'notionalCap': '100000000',
                             'notionalFloor': '40000000'},
                    'maintenanceMarginRate': 0.1,
                    'maxLeverage': 5.0,
                    'maxNotional': 100000000.0,
                    'minNotional': 40000000.0,
                    'tier': 6},
                   {'currency': 'USDT',
                    'info': {'bracket': '7',
                             'cum': '4703800',
                             'initialLeverage': '4',
                             'maintenanceMarginRate': '0.125',
                             'notionalCap': '200000000',
                             'notionalFloor': '100000000'},
                    'maintenanceMarginRate': 0.125,
                    'maxLeverage': 4.0,
                    'maxNotional': 200000000.0,
                    'minNotional': 100000000.0,
                    'tier': 7},
                   {'currency': 'USDT',
                    'info': {'bracket': '8',
                             'cum': '9703800',
                             'initialLeverage': '3',
                             'maintenanceMarginRate': '0.15',
                             'notionalCap': '400000000',
                             'notionalFloor': '200000000'},
                    'maintenanceMarginRate': 0.15,
                    'maxLeverage': 3.0,
                    'maxNotional': 400000000.0,
                    'minNotional': 200000000.0,
                    'tier': 8},
                   {'currency': 'USDT',
                    'info': {'bracket': '9',
                             'cum': '49703800',
                             'initialLeverage': '2',
                             'maintenanceMarginRate': '0.25',
                             'notionalCap': '600000000',
                             'notionalFloor': '400000000'},
                    'maintenanceMarginRate': 0.25,
                    'maxLeverage': 2.0,
                    'maxNotional': 600000000.0,
                    'minNotional': 400000000.0,
                    'tier': 9},
                   {'currency': 'USDT',
                    'info': {'bracket': '10',
                             'cum': '199703800',
                             'initialLeverage': '1',
                             'maintenanceMarginRate': '0.5',
                             'notionalCap': '1000000000',
                             'notionalFloor': '600000000'},
                    'maintenanceMarginRate': 0.5,
                    'maxLeverage': 1.0,
                    'maxNotional': 1000000000.0,
                    'minNotional': 600000000.0,
                    'tier': 10}]}
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**